Projects/UI/HGDesignSystem/Sources/Components/TextField/HGTextField.swift

```diff
@@ -0,0 +1,202 @@
+        let state = makeState()
+        
+        return HStack(spacing: 8) {
+            TextField("", text: $text)
+                .focused($isFocused)
+                .foregroundStyle(.gray95)
+                .setTypo(size.font)
+                .frame(height: 24)
+                .background(alignment: .leading) {
+                    if text.isEmpty {
+                        Text(placeholder)
+                            .setTypo(size.font)
+                            .foregroundStyle(HGColors.gray30)
+                    }
+                }
+            
+            if !hiddenClearButton {
+                Button {
+                    withAnimation {
+                        text.removeAll()
+                    }
+                } label: {
+                    HGIcons.close.image // TODO: #34 병합 이후 다른 아이콘으로 변경 예정
+                        .resizable()
+                        .frame(24)
+                        .foregroundStyle(HGColors.gray40)
+                }
+            }
+        }
+        .frame(maxWidth: .infinity)
+        .padding(.bottom, 6)
+        .overlay(alignment: .bottom) {
+            Rectangle()
+                .fill(underLineColor(state: state))
+                .frame(height: 1)
+        }
+    }
+    
+    private var helperArea: some View {
+        HStack(spacing: 8) {
+            if let errorMessage {
+                Text(errorMessage)
+                    .setTypo(.caption_12_medium)
+                    .foregroundStyle(.redMain)
+            }
+            Spacer()
+            if let maxCount {
+                Text("\(text.count) / \(maxCount)")
+                    .setTypo(.caption_12_medium)
+                    .foregroundStyle(.gray50)
+                    .if(maxCount >= text.count) {
+                        $0
+                            .contentTransition(.numericText(countsDown: true))
+                            .animation(.default, value: text)
+                    }
+            }
+        }
+        .padding(.top, 4)
+    }
+}
+
+// MARK: - Private Method
+private extension HGTextField {
+    func makeState() -> TextFieldState {
+        if !self.isEnabled {
+            return .disabled
+        }
+        
+        if self.isFocused {
+            return .focused
+        }
+        
+        return .normal
+    }
+    
+    func underLineColor(state: TextFieldState) -> Color {
+        switch state {
+        case .normal, .disabled, .error:
+            return HGColors.gray40.color
+        case .focused:
+            return HGColors.orange500Main.color
+        }
+    }
+}
+
+public extension HGTextField {
+    /// 텍스트필드 위에 텍스트필드 제목을 설정합니다.
+    /// - parameter title: 이 텍스트 필드의 제목
+    /// - parameter required: 이 텍스트 필드의 입력 필수 여부
+    func setTitle(_ title: String?, required: Bool? = nil) -> Self {
+        var copy = self
+        copy.title = title
+        if let required { copy.required = required }
+        
+        return copy
```

Comment:
뷰 프로퍼티가 옵셔널아니라서 nil말고 false를 디폴트로 써도 좋을거같아서 적어봅니당

```suggestion
    func setTitle(_ title: String?, required: Bool = false) -> Self {
        var copy = self
        copy.title = title
        copy.required = required
        
        return copy
```